astropy/convolution/convolve.py

```diff
@@ -724,11 +721,14 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
     if interpolate_nan:
         rifft = (ifftn(fftmult)) / bigimwt
         if not np.isscalar(bigimwt):
-            rifft[bigimwt < min_wt] = np.nan
-            if min_wt == 0.0:
-                rifft[bigimwt == 0.0] = 0.0
+            if min_wt > 0.:
+                rifft[bigimwt < min_wt] = np.nan
+            else:
+                # Set anything with no weight to zero (taking into account
+                # slight offsets due to floating-point errors).
+                rifft[np.abs(bigimwt) < 10 * np.finfo(bigimwt.dtype).eps] = 0.0
```

Comment:
I think this *should* work, but it might be safer to remove `np.abs`?  Negative weights are not defined, so they should also be zero'd here.

Reply:
Yes, makes sense. Done. Also made the `floatclose` depend on the actual values being compared, as there were still failing tests.